locator/topology.hh

```diff
@@ -12,37 +12,128 @@
 
 #include <unordered_set>
 #include <unordered_map>
+#include <iostream>
 
 #include <seastar/core/future.hh>
 #include <seastar/core/sstring.hh>
 #include <seastar/util/bool_class.hh>
 
 #include "locator/types.hh"
 #include "inet_address_vectors.hh"
-#include "utils/fb_utilities.hh"
 
 using namespace seastar;
 
 namespace locator {
 
+class topology;
+
+class node : public enable_lw_shared_from_this<node> {
```

Comment:
Why is shared_ptr needed? Do not objects outlive topology objects?

If not, the topology pointers are sufficient to keep them alive.

Reply:
I used them to share the node objects across topology instances, otherwise they'll need to be copied (not too bad)
and indeed pointers to nodes will be valid as long the the e_r_m -> token_metadata -> topology chain is valid.

Reply:
using `unique_ptr<node>` in v8